util/dvsim/SgeLauncher.py

```diff
@@ -0,0 +1,174 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+# ------------------------------------
+#      SgeLauncher Class
+#
+# ------------------------------------
+import os
+import shlex
+import subprocess
+import SGE
+from subprocess import PIPE, Popen
+from Launcher import ErrorMessage, Launcher, LauncherError
+global job_name
+
+pid = os.getpid()
+
+
+class SgeLauncher(Launcher):
+    """
+    Implementation of Launcher to launch jobs in the user's local workstation.
+    """
+
+    # Misc common SgeLauncher settings.
+    max_odirs = 5
+
+    def __init__(self, deploy):
+        '''Initialize common class members.'''
+
+        super().__init__(deploy)
+
+        # Popen object when launching the job.
+        self.process = None
+
+    def _do_launch(self):
+        global job_name
+        # Update the shell's env vars with self.exports. Values in exports must
+        # replace the values in the shell's env vars if the keys match.
+        exports = os.environ.copy()
+        if self.deploy.exports:
+            exports.update(self.deploy.exports)
+
+        # Clear the magic MAKEFLAGS variable from exports if necessary. This
+        # variable is used by recursive Make calls to pass variables from one
+        # level to the next. Here, self.cmd is a call to Make but it's
+        # logically a top-level invocation: we don't want to pollute the flow's
+        # Makefile with Make variables from any wrapper that called dvsim.
+        if 'MAKEFLAGS' in exports:
+            del exports['MAKEFLAGS']
+
+        self._dump_env_vars(exports)
+
+        try:
+            f = open(self.deploy.get_log_path(),
+                     "w",
+                     encoding="UTF-8",
+                     errors="surrogateescape")
+            f.write("[Executing]:\n{}\n\n".format(self.deploy.cmd))
+            f.flush()
+            # ---------- prepare SGE job struct -----
+            sgeJob = SGE.qsubOptions()
+            sgeJob.args.N = 'VCS_RUN_' + str(pid)  # Name of Grid Engine job
+            if "build.log" in self.deploy.get_log_path():
+                sgeJob.args.N = 'VCS_BUILD_' + str(pid)  # Name of Grid Engine job
+
+            job_name = sgeJob.args.N
+            sgeJob.args.t = '0'      # Define an array job with 20 subjobs
```

Comment:
This and line 70 below have the same comment? Something is not right...

Reply:
Updated the comment.